yaperos/app-nodejs-codechallenge
# Enhancing Transaction Processing and Anti-Fraud Validation

Hello there! :wave:

I'm excited to submit my solution for the Yape Code Challenge as a Backend Engineer (NodeJS). This pull request contains enhancements to the transaction processing and anti-fraud validation components, ensuring seamless handling of financial transactions while adhering to the specified requirements.

Changes Made
Implemented enhanced transaction processing logic:
- Created a new endpoint for creating transactions with improved request structure.
- Integrated anti-fraud validation for every transaction, ensuring secure and reliable processing.
- Handled different transaction statuses (pending, approved, rejected) based on anti-fraud evaluation.
- Implemented logic to reject transactions with a value greater than 1000, as per the challenge instructions.

Updated GraphQL API:
- Extended the GraphQL API to support the new transaction structure and status handling.
- Integrated the enhanced transaction processing flow into the existing GraphQL schema.

How to Test
- Make sure to have Docker and docker-compose installed.
- Run the following command to start the project: **bash start.sh**
- Wait a few seconds for the services to initialize.
- Utilize the provided Postman collections to test the REST and GraphQL APIs.

Your feedback is greatly appreciated, and I look forward to discussing my implementation with you. Thank you for this exciting opportunity! 🚀 

Best regards,
Cristobal Contreras
